### PR TITLE
Some mimic changes

### PR DESCRIPTION
**Pull request by loqoman**
([I WAS HIDING](https://www.youtube.com/watch?v=Fm8PkRWTnns))

:cl: loqoman
tweak: Mimics have been tweaked a bit to be more like...Mimics!
/:cl:

More of a learning project if anything, this is going to be a *slight* rework of mimics!

What has been done:

- Mimic's now have an ability that changes their sprite, and has them lay dormant (But they can still move) 
- Mimic's have some slightly better sprites for when they are aggressive, as well as sprites for the ability.

What has yet to be done:

- Finishing the mimic ability. The final idea is for the ability is to have the mimic disguise as a crate, and when crewmembers interact with the "Crate" the mimic will go full rage mode, and deal a large amount of damage to the person that touched it.
- Various cleanup(Making the visible_message nicer, cleaning up sprite timings, ect.)

As you can see, this is not a completed project, but considering this is my first contribution, I want to get it out there for QA/QC, as well as additional design ideas if people have them. 
image 
(Sprite)

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/mimic.dm [hunk 1]
```diff
@@ -73,8 +77,8 @@
 					"<span class='userdanger'>\The [src] knocks you down!</span>")
 
 /mob/living/simple_animal/hostile/mimic/crate/proc/trigger()
-	if(!attempt_open)
-		visible_message("<b>[src]</b> starts to move!")
+	if(!attempt_open) //Have not attempted to open
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/hostile/mimic.dm]: comments like this are worthless, remove all comments you've added.

File: code/modules/mob/living/simple_animal/hostile/mimic.dm [hunk 2]
```diff
@@ -266,3 +270,22 @@ GLOBAL_LIST_INIT(protected_objects, list(/obj/structure/table, /obj/structure/ca
...
+
+/datum/action/innate/mimic/root
+	name = "Root"
+	button_icon_state = "mimicA"
+	desc = "Roots yourself in place and disguises you as a normal crate. Anyone to try to open you up is given a nasty surprise!"
+
+/datum/action/innate/mimic/root/Activate()
+	var/mob/living/simple_animal/hostile/mimic/crate/S = owner
+
+	if(S.attempt_open == TRUE) //If we have tried to open it
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/hostile/mimic.dm]: just ```if(S.attempt_open)``` is preferred.

File: code/modules/mob/living/simple_animal/hostile/mimic.dm [hunk 3]
```diff
@@ -266,3 +270,22 @@ GLOBAL_LIST_INIT(protected_objects, list(/obj/structure/table, /obj/structure/ca
...
+		S.visible_message("<span class='danger'><b>[S]</b> disguises itself as a crate</span>")
+
+		S.attempt_open = FALSE	//I WAS HIDING
+		S.icon_state = "crate"
+		S.icon_living = "crate"
+
+
+	//TBD, but see trellio
+	return
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/hostile/mimic.dm]: a return at the end of a function that does nothing is a trailing return.
It's unneeded, and should be removed.

File: code/modules/mob/living/simple_animal/hostile/mimic.dm [hunk 4]
```diff
@@ -266,3 +270,22 @@ GLOBAL_LIST_INIT(protected_objects, list(/obj/structure/table, /obj/structure/ca
 		return
 	icon_state = TrueGun.icon_state
 	icon_living = TrueGun.icon_state
+
+/datum/action/innate/mimic/root
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/hostile/mimic.dm]: this shouldn't be ```/innate```

File: code/modules/mob/living/simple_animal/hostile/mimic.dm [hunk 5]
```diff
@@ -266,3 +270,22 @@ GLOBAL_LIST_INIT(protected_objects, list(/obj/structure/table, /obj/structure/ca
 		return
 	icon_state = TrueGun.icon_state
 	icon_living = TrueGun.icon_state
+
+/datum/action/innate/mimic/root
+	name = "Root"
+	button_icon_state = "mimicA"
+	desc = "Roots yourself in place and disguises you as a normal crate. Anyone to try to open you up is given a nasty surprise!"
+
+/datum/action/innate/mimic/root/Activate()
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/hostile/mimic.dm]: this should be ```Trigger()```, with the first line of it being:
```
if(!..())
    return 0
```

**Comment by InsaneHyena** [comment]
We have mimics?

**Comment by Mey-Ha-Zah** [comment]
Thought they were called Morphs.

**Comment by RemieRichards** [comment]
Morphs != Mimics.
Two completely different sections of code.

**Comment by Supermichael777** [comment]
mimics are just a tragically underused fake crate.

**Comment by Bawhoppen** [comment]
I wanted to add loot drops to them though and now you walk in and do this

**Comment by loqoman** [comment]
@RemieRichards It's a very basic simple mob that acts as a fake crate, and triggers when people attempt to open it.

**Comment by loqoman** [comment]
morps are the mob that eats things to disguise as them

File: code/modules/mob/living/simple_animal/hostile/mimic.dm (hunk 1 above)
Comment (loqoman) [reply code/modules/mob/living/simple_animal/hostile/mimic.dm]: Alright, what kind of comments SHOULD I be writing?

**Comment by RemieRichards** [comment]
I Know?? I'm telling THEM that.

**Comment by loqoman** [comment]
Eh? Oh sorry, read it wrong. MB

File: code/modules/mob/living/simple_animal/hostile/mimic.dm (hunk 4 above)
Comment (loqoman) [reply code/modules/mob/living/simple_animal/hostile/mimic.dm]: Okay, what types of actions are supposed to be innate?

**Comment by loqoman** [comment]
@Bawhoppen What loot drops were you thinking about?

**Comment by Bawhoppen** [comment]
Well nothing super specific, but I just wanted them to drop something lootwise to fit with the theme of how mimics sometimes have loot inside them after you slay them in other lore,

**Comment by Armhulen** [comment]
i agree with you on the loot thing but it's a xenobio mob so you can't really do that.

**Comment by Armhulen** [comment]
also you can't make a mob work like an object so why not have root make an object that when interacted with opens and does the whole mimic deal

**Comment by WJohn** [comment]
This sprite is terrible but at least there is now a clear indication it is a minimal. Sprite can be improved later.

**Comment by loqoman** [comment]
I'm a lazy fuck!

**Pull request closed without merging**